### PR TITLE
SYNCT-191: Made usage of REST client generic

### PR DESCRIPTION
**Pull request by areklalo**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/sync2/api/model/enums/Resources.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
     PATIENT("patient"),
     VISIT("visit"),
     ENCOUNTER("encounter"),
-    OB("ob"),
+    OB("observation"),
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/sync2/api/model/enums/Resources.java]:
```suggestion
    OB(SyncCategoryConstants.CATEGORY_OBSERVATION),
```

File: api/src/test/java/org/openmrs/module/sync2/client/rest/RESTClientHelperTest.java [hunk 2]
```diff
@@ -53,23 +79,41 @@ public void deleteRequest() throws Exception {
...
+		String json = getJson(object);
+		return SimpleObject.parseJson(json);
+	}
+
+	private String getJson(Object object) {
+		Gson defaultJsonParser = createDefaultGson();
+		return defaultJsonParser.toJson(object);
+	}
+
+	private Gson createDefaultGson() {
```
Comment (wojtekbu) [inline api/src/test/java/org/openmrs/module/sync2/client/rest/RESTClientHelperTest.java]: Please move createDefaultGson to constants and use the same Gson in the business logic and in tests

File: api/src/main/java/org/openmrs/module/sync2/client/SimpleObjectMessageConverter.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			throw new HttpMessageNotWritableException("Could not serialize object. Msg: " + e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * This method configures Gson. We need to use workaround for null dates.
+	 *
+	 * @return definitive null safe Gson instance
+	 */
+	private Gson createDefaultGson() {
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/sync2/client/SimpleObjectMessageConverter.java]: Please move createDefaultGson to constants and use the same Gson in the business logic and in tests

File: api/src/test/java/org/openmrs/module/sync2/api/sync/SyncClientTest.java [hunk 4]
```diff
@@ -60,7 +60,7 @@
     private static final String PATIENT_CATEGORY = "patient";
     private static final String VISIT_CATEGORY = "visit";
     private static final String ENCOUNTER_CATEGORY = "encounter";
-    private static final String OB_CATEGORY = "ob";
+    private static final String OB_CATEGORY = "observation";
```
Comment (wojtekbu) [inline api/src/test/java/org/openmrs/module/sync2/api/sync/SyncClientTest.java]: Please use SyncCategoryConstants.CATEGORY_OBSERVATION

Comment (areklalo) [reply api/src/test/java/org/openmrs/module/sync2/api/sync/SyncClientTest.java]: Changed.

File: api/src/main/java/org/openmrs/module/sync2/client/SimpleObjectMessageConverter.java (hunk 3 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/sync2/client/SimpleObjectMessageConverter.java]: Done

File: api/src/test/java/org/openmrs/module/sync2/client/rest/RESTClientHelperTest.java (hunk 2 above)
Comment (areklalo) [reply api/src/test/java/org/openmrs/module/sync2/client/rest/RESTClientHelperTest.java]: Done

**Pull request merged**